parsers/gdscript.c

```diff
@@ -0,0 +1,1346 @@
+
+		if (arglist)
+			e.extensionFields.signature = vStringValue (arglist);
+
+		return makeTagEntry (&e);
+	}
+
+	return CORK_NIL;
+}
+
+static int makeSimpleGDScriptTag (const tokenInfo *const token, gdscriptKind const kind)
+{
+	if (GDScriptKinds[kind].enabled)
+	{
+		tagEntryInfo e;
+
+		initGDScriptEntry (&e, token, kind);
+		return makeTagEntry (&e);
+	}
+
+	return CORK_NIL;
+}
+
+static int makeSimpleGDScriptRefTag (const tokenInfo *const token,
+                                   gdscriptKind const kind,
+                                   int roleIndex, xtagType xtag)
+{
+	if (isXtagEnabled (XTAG_REFERENCE_TAGS))
+	{
+		tagEntryInfo e;
+
+		initRefTagEntry (&e, vStringValue (token->string),
+		                 kind, roleIndex);
+
+		e.lineNumber	= token->lineNumber;
+		e.filePosition	= token->filePosition;
+
+		if (xtag != XTAG_UNKNOWN)
+			markTagExtraBit (&e, xtag);
+
+		return makeTagEntry (&e);
+	}
+
+	return CORK_NIL;
+}
+
+static void *newPoolToken (void *createArg CTAGS_ATTR_UNUSED)
+{
+	tokenInfo *token = xMalloc (1, tokenInfo);
+	token->string = vStringNew ();
+	return token;
+}
+
+static void deletePoolToken (void *data)
+{
+	tokenInfo *token = data;
+	vStringDelete (token->string);
+	eFree (token);
+}
+
+static void clearPoolToken (void *data)
+{
+	tokenInfo *token = data;
+
+	token->type			= TOKEN_UNDEFINED;
+	token->keyword		= KEYWORD_NONE;
+	token->indent		= 0;
+	token->lineNumber   = getInputLineNumber ();
+	token->filePosition = getInputFilePosition ();
+	vStringClear (token->string);
+}
+
+static void copyToken (tokenInfo *const dest, const tokenInfo *const src)
+{
+	dest->lineNumber = src->lineNumber;
+	dest->filePosition = src->filePosition;
+	dest->type = src->type;
+	dest->keyword = src->keyword;
+	dest->indent = src->indent;
+	vStringCopy(dest->string, src->string);
+}
+
+/* Skip a single or double quoted string. */
+static void readString (vString *const string, const int delimiter)
+{
+	int escaped = 0;
+	int c;
+
+	while ((c = getcFromInputFile ()) != EOF)
+	{
+		if (escaped)
+		{
+			vStringPut (string, c);
+			escaped--;
+		}
+		else if (c == '\\')
+			escaped++;
+		else if (c == delimiter || c == '\n' || c == '\r')
+		{
+			if (c != delimiter)
+				ungetcToInputFile (c);
+			break;
+		}
+		else
+			vStringPut (string, c);
+	}
+}
+
+/* Skip a single or double triple quoted string. */
+static void readTripleString (vString *const string, const int delimiter)
+{
+	int c;
+	int escaped = 0;
+	int n = 0;
+	while ((c = getcFromInputFile ()) != EOF)
+	{
+		if (c == delimiter && ! escaped)
+		{
+			if (++n >= 3)
+				break;
+		}
+		else
+		{
+			for (; n > 0; n--)
+				vStringPut (string, delimiter);
+			if (c != '\\' || escaped)
+				vStringPut (string, c);
+			n = 0;
+		}
+
+		if (escaped)
+			escaped--;
+		else if (c == '\\')
+			escaped++;
+	}
+}
+
+static void readIdentifier (vString *const string, const int firstChar)
+{
+	int c = firstChar;
+	do
+	{
+		vStringPut (string, (char) c);
```

Comment:
The type of the second parameter of vStringPut is `int`. So you don't have to convert the type of `c` to `char` here.

Reply:
Please, ignore above comment. I found it came from the python parser. I will fix both python and gdscript parsers together after merging gdscript.

Reply:
I think I haven't touched this particular function, checked the python parser and it also has the little extra cast. Defies expectation a little. Almost done with going through the code coverage moans, this should put it over the threshold.

Reply:
Oh I just realised you can click through to see more code coverage complaints, I have a lot more lines to look at

Reply:
Alright, got a little hard to match line numbers near the end as code changed but this should surely get us above the threshold